src/test/compile-fail/E0049.rs

```diff
@@ -16,6 +16,7 @@ struct Bar;
 
 impl Foo for Bar {
     fn foo(x: bool) -> Self { Bar } //~ ERROR E0049
+                                    //~| but got 0 type parameters
```

Comment:
You're not testing for the expected part in the above.  Might as well throw that in, too.

Since you add a few more label cases, you should also make sure to test your new cases in the unit test.